source/server/server.cc

```diff
@@ -545,10 +545,14 @@ void InstanceImpl::initialize(const Options& options,
     stats_store_.addSink(*sink);
   }
 
-  // Some of the stat sinks may need dispatcher support so don't flush until the main loop starts.
-  // Just setup the timer.
-  stat_flush_timer_ = dispatcher_->createTimer([this]() -> void { flushStats(); });
-  stat_flush_timer_->enableTimer(config_.statsFlushInterval());
+  if (!config_.statsFlushOnAdmin()) {
+    // Some of the stat sinks may need dispatcher support so don't flush until the main loop starts.
+    // Just setup the timer.
+    stat_flush_timer_ = dispatcher_->createTimer([this]() -> void { flushStats(); });
+    stat_flush_timer_->enableTimer(config_.statsFlushInterval());
+  } else {
+    stat_flush_timer_.reset();
```

Comment:
Can initialize() be called twice? Because if so I think we'll be adding the sinks a second time in lines 544-546 above. And i fit can only be called once I think stat_flush_timer_ cannot be populated so there should be no need to reset it, though you could ASSERT that.

Reply:
This makes sense to me. The `reset` was more of a sanity thing but the ASSERT makes more sense